src/main/java/com/iexec/core/task/TaskRepository.java

```diff
@@ -24,7 +24,7 @@
 import java.util.List;
 import java.util.Optional;
 
-interface TaskRepository extends MongoRepository<Task, String> {
+public interface TaskRepository extends MongoRepository<Task, String> {
```

Comment:
Why public?

Reply:
It is used by `TaskUpdateManager`, which I moved into the `com.iexec.core.task.update` package - where it should have been. A package-private visibility is insufficient, then.

Reply:
Shouldn't `TaskRepository` be accessed through `TaskService` in this case ?

Reply:
This could allow to remove TaskRepository as a dependency from TaskUpdateManager, which can be a good idea because there are lots of injected dependency, which is not recommended.
See code quality reports and **Methods should not have too many parameters** rule.

Reply:
You're right. I'll work on that.

Reply:
Done in 100df2d.